From 6f46c3c3b007f1aed77bbb4d1657fab8521e2e08 Mon Sep 17 00:00:00 2001
From: Rich Salz <rsalz@openssl.org>
Date: Thu, 11 Sep 2014 13:08:30 -0400
Subject: [PATCH] RT2772 update: c_rehash was broken

Move the readdir() lines out of the if statement, so
that flist is available globally.

Reviewed-by: Tim Hudson <tjh@openssl.org>
---
 tools/c_rehash.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/tools/c_rehash.in
+++ b/tools/c_rehash.in
@@ -97,10 +97,10 @@ sub hash_dir {
 	my %hashlist;
 	print "Doing $_[0]\n";
 	chdir $_[0];
+	opendir(DIR, ".");
+	my @flist = readdir(DIR);
+	closedir DIR;
 	if ( $removelinks ) {
-		opendir(DIR, ".");
-		my @flist = readdir(DIR);
-		closedir DIR;
 		# Delete any existing symbolic links
 		foreach (grep {/^[\da-f]+\.r{0,1}\d+$/} @flist) {
 			if (-l $_) {
